src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -435,7 +436,8 @@ void Compressor<HttpMethodMetadata, HttpMethodCompressor>::EncodeWith(
           Slice::FromStaticString(":method"), Slice::FromStaticString("PUT"));
       break;
     case HttpMethodMetadata::ValueType::kInvalid:
-      Crash("invalid http method encoding");
+      LOG(ERROR) << "Not encoding bad http method";
```

Comment:
Just to confirm - is it okay to continue here without crashing? Could this put us in an unsafe state?

Reply:
The worst is that we'll send a bad request that will get rejected later